src/proto/net.proto

```diff
@@ -3,17 +3,31 @@ syntax = "proto2";
 package proto;
 
 message ServerMessage {
+  optional bool status = 1;
+
   oneof message_type {
-    string message = 1;
+    string message = 2;
+    Construct build_update = 3;
   }
 }
 
 message ClientMessage {
+  optional bool status = 1;
+  
   oneof message_type {
-    string message = 1;
+    string message = 2;
+    Construct build_request = 3;
   }
 }
 
+message Construct {
+  optional int32 type = 1;
```

Comment:
Note: we will eventually need unique identifiers to sync up instance-specific updates between client and server. For example, trap activation when there are multiple instances of the same trap.